.codegen/impl.go.tmpl

```diff
@@ -13,22 +15,71 @@ import (
 	{{range .ImportedPackages}}
 	"github.com/databricks/databricks-sdk-go/service/{{.}}"{{end}}
 )
-{{range .Services}} {{if not .IsDataPlane}}
+{{range .Services}}
 // unexported type that holds implementations of just {{.Name}} API methods
 type {{.CamelName}}Impl struct {
+	{{- if .IsDataPlane}}
+	oauth2.DataPlaneHelper
+	controlPlane *{{.ControlPlaneService.PascalName}}API
+	{{end -}}
 	client *client.DatabricksClient
 }
 
 {{range .Methods}}
+{{if .Service.IsDataPlane}}
+func (a *{{.Service.CamelName}}Impl) {{.PascalName}}(ctx context.Context{{if .Request}}, request {{.Request.PascalName}}{{end}}) {{ template "response-type" . }} {
+	getRequest := {{.Service.DataPlaneInfoMethod.Request.PascalName}}{
+	{{- range .Service.DataPlaneInfoMethod.Request.Fields}}
+		{{.PascalName}}: request.{{.PascalName}},
+	{{end}}
+	}
+	token, err := a.client.Config.GetToken()
+	if err != nil {
+		return nil, err
+	}
+	infoGetter := func() (*oauth2.DataPlaneInfo, error) {
+		response, err := a.controlPlane.{{.Service.DataPlaneInfoMethod.PascalName}}(ctx, getRequest)
+		if err != nil {
+			return nil, err
+		}
+		if response.{{(index .DataPlaneInfoFields 0).PascalName}} == nil {
+			return nil, errors.New("resource does not support direct Data Plane access")
```

Comment:
When would this case happen? Is this a bug in the API definition? Or is it possible that some model serving endpoints support direct-to-dataplane access and others don't?

Reply:
DataPlane access needs to be enabled on a resource level. If you call this method for a Model which is not "optimized" (their flag for DataPlane), this field won't be set (the endpointURL does not exists)